engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -713,6 +717,11 @@ private void execute() throws YarnException, IOException
         expiryTime = StramUserLogin.refreshTokens(tokenLifeTime, FileUtils.getTempDirectoryPath(), applicationId, conf, hdfsKeyTabFile, credentials, rmAddress, true);
       }
 
+      if (System.currentTimeMillis() - nodeReportUpdateTime >= UPDATE_NODE_REPORTS_INTERVAL) {
```

Comment:
We should see to not retrieve currentTimeMillis in one iteration (also see security stuff above).